test/release/examples/primers/interopWithC.chpl

```diff
@@ -0,0 +1,332 @@
+   ``chpl_library_init``, which is defined in
+   ``$CHPL_HOME/runtime/include/chpl-init.h`` and accessible with the above
+   ``#include``.
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 5-7
+*/
+
+/*   
+   Then, if the exported functions rely upon any global variables, the module
+   initialization function declared in the header file must be called.  If there
+   is uncertainty about whether doing so is necessary, it is recommended to call
+   this function.  In this example, this call occurs like this:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 8-9
+*/
+
+//
+// After that, the exported functions may be called:
+//
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 11-13
+*/
+
+/*
+   When ending the C program, the user must explicitly shut down the Chapel
+   runtime and module code.  This is done by calling ``chpl_library_finalize``:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 15-19
+*/
+
+/*
+   Once shut down, the Chapel runtime cannot be restarted in the current C
+   program.
+*/
+
+/*
+   .. _primers-C-interop-compiling-2:
+
+   Compiling C Code with the Chapel Library
+   ++++++++++++++++++++++++++++++++++++++++
+*/
+
+/*
+   Compiling C code with the generated Chapel library file is generally complex,
+   but can be made simpler through the use of the ``--library-makefile`` flag
+   as described in :ref:`readme-libraries-linking`.
+   
+   An example of compiling a C program with a generated Chapel library using the
+   generated Makefile can be found in the `Makefile
+   <https://github.com/chapel-lang/chapel/blob/master/test/release/examples/primers/Makefile>`_
+   for the primers directory, to build this source file.
+*/ 
+
+/*
+   .. _primers-C-interop-using-C:
+
+   Using C Code in Chapel
+   ----------------------
+*/
+
+/*
+   Chapel has support for C code but we need to tell the compiler
+   about the C symbols using the ``extern`` keyword
+*/
+
+   extern proc baz(): int;
+
+//
+// The function can then be called as normal:
+//
+   proc callsBaz() {
+      writeln(baz());
+   }
+
+//
+// Such functions can even be called in exported functions:
+//
+
+   export proc alsoCallsBaz() {
+      writeln(baz());
+      callsBaz();
+   }
+
+// You can tell the chapel compiler where to look for these C functions by adding a
+// require statement
+   require "cHelper.h", "cHelper.c";
+// You must inlcude both a header file and C file.
+// Chapel also supports require statements for ``.o`` files and
+// for archived libraries using the ``-l`` flag
+/*
+   .. code-block:: chapel
+
+         require "foo.o", "-lapache-arrow";
+*/
+
+// Alternatively you can also include their names while invoking the chapel compiler.
```

Comment:
I think this should have a `.. code-block:: bash` example to go with it

Reply:
I agree with Lydia here: please add an example invocation.  When I was playing with the interop module, I guessed there were two things I needed to figure out: (1) how to write the Chapel code to declare and call the C functions and (2) how to invoke the compiler. I think your primer makes it pretty easy to figure out (1) and I don't think it's that hard to figure out (2), but someone impatiently skimming the primer to figure (2) out might just ctrl+f for "chpl" to see an example of the command.